src/backend/cdb/cdbpartition.c

```diff
@@ -2768,15 +2768,16 @@ partition_policies_equal(GpPolicy *p, PartitionNode *pn)
 			}
 			else
 			{
-				if (p->attrs == 0)
-					/* random policy, skip */
-					;
```

Comment:
Maybe you should have just added continue here?
```c
					continue;
```

Reply:
There's a whole block at https://github.com/arenadata/gpdb/blob/a6e09d63a5e0913ebf08ec204c27edd9f1476560/src/backend/cdb/cdbpartition.c#L2782
and I'm not sure that it should be skipped if !p->attrs